### DISCUSSION TITLE
CSPM API list assets

### DISCUSSION THREAD

**Discussion by notdodo** [CrowdStrike/falconpy#1060] [Ideas]
It seems that it's not possible to have an API to fetch the list of all cloud assets (from the web UI: cloud-security/asset-inventory/cloud-assets).

HTTP call to get the list of asset IDs from UI: `/api2/cspmregistration/settings/queries/assets/v1?filter=&offset=20&limit=20`
To retrieve the assets information from UI : `/api2/cspmregistration/settings/entities/assets/v1?ids=[listOfIDs]`

Both these HTTP calls are only available via web UI **but not on APIs.**

Related IDEA: https://eu-1.ideas.crowdstrike.com/ideas/IDEA-I-9984

**Comment by jshcodes** [comment]
Just confirmed that new API operations for providing this functionality are on the roadmap and should be available soon! 😃

**Reply by notdodo** [reply]
Nice! Thank you @jshcodes; do you have an estimate of when these APIs should be available (EU-1 customer)?

**Reply by jshcodes** [reply]
This is slated for release sometime early next year (Q1 or Q2).  😎 

> Reminder: Roadmap release dates are tentative and subject to change.
